File: cfgs/java_cfgs_foo/NEXT_POWER_OF_2.java
Nodes
1;special;;;
2;statement;int count = 0;14;
3;conditional;(foo_1 > 0) && ((foo_2 & (foo_3 - 1)) == 0);15;
4;statement;return foo_1;15;
5;exit;;;
6;conditional;foo_1 != 0;16;
7;statement;n >>= 1;17;
8;statement;count += 1;18;
9;statement;return 1 << foo_1;20;
Edges
0;2;
1;3;
2;4;6;
3;5;
4;
5;9;7;
6;8;
7;6;
8;5;

File: cfgs/pythoncfgs/NEXT_POWER_OF_2.py
Nodes
1;statement;foo_1 = 0;8;
2;conditional;foo_1 and (not foo_2 & foo_3 - 1);9;
3;conditional;foo_1 != 0;11;
4;statement;return 1 << foo_1;14;
5;statement;foo_1 >>= 1;12;
6;statement;foo_1 += 1;13;
7;statement;return foo_1;10;
8;exit;;
Edges
0;2;
1;3;7;
2;4;5;
3;8;
4;6;
5;3;
6;8;
7;
